pkg/registry/apis/provisioning/jobs/exporter.go

```diff
@@ -56,25 +67,71 @@ func (r *exporter) Export(ctx context.Context,
 	progress func(provisioning.JobStatus) error,
 ) (*provisioning.JobStatus, error) {
 	logger := logging.FromContext(ctx)
-	dashboardIface := r.client.Resource(schema.GroupVersionResource{
-		Group:    "dashboard.grafana.app",
-		Version:  "v2alpha1",
-		Resource: "dashboards",
-	})
-
 	err := progress(provisioning.JobStatus{
 		State:   provisioning.JobStateWorking,
 		Message: "getting folder tree...",
 	})
 	if err != nil {
 		return nil, err
 	}
+	var ref string
+	if r.repository.Config().Spec.Type == provisioning.GitHubRepositoryType {
+		ref = r.repository.Config().Spec.GitHub.Branch
+	}
+	ns := r.repository.Config().GetNamespace()
+	logger = logger.With("ref", ref, "namespace", ns)
 
-	folders, err := resources.FetchRepoFolderTree(ctx, r.client)
+	err = progress(provisioning.JobStatus{
+		State:   provisioning.JobStateWorking,
+		Message: "exporting folders...",
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: handle pagination
+	rawFolders, err := r.folders.List(ctx, metav1.ListOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("failed to list folders: %w", err)
 	}
 
+	unprovisionedFolders := make([]unstructured.Unstructured, 0)
```

Comment:
nit: idiomatic is to just use a nil:
```suggestion
	var unprovisionedFolders []unstructured.Unstructured
```
or with a cap hint
```suggestion
	unprovisionedFolders := make([]unstructured.Unstructured, len(rawFolders.Items) / 2)
```

Reply:
I am actually more in favour of what you suggested. I think it was Copilot talking for me :)